worker/index.js

```diff
@@ -70,8 +70,29 @@ async function work () {
     }
   }
 
+  async function subWrapper (sub, event, fn) {
+    sub.on(event, async (...args) => {
+      console.log(`event ${event} triggered with args`, args)
+      try {
+        await fn(...args)
+      } catch (error) {
+        console.error(`error running ${event}`, error)
+        return
+      }
+      console.log(`finished ${event}`)
+    })
+    sub.on('error', () => {
+      // LND connection lost
+      // see https://www.npmjs.com/package/ln-service#subscriptions
+      sub.removeAllListeners()
```

Comment:
> Besides the events listed in this documentation, all EventEmitter objects returned by any of the subscribeTo... methods also offer an error event. This event is triggered e.g. when LND is shut down. So, in order to detect and correctly handle such a situation, robust client code should generally also add a listener for the error event.

-- https://www.npmjs.com/package/ln-service#subscribetoinvoices

So iiuc, this should only trigger if LND is shutting down. Afaict, we don't need any "reconnect code". But to be sure, I would run this in parallel with the old code since ~~they don't interfere with each other. If the subscription was faster than the poll, the polling code will just see in the next poll that the invoice was already confirmed.~~

Ok, just realized they _do_ interfere with each other. I tested push notifications and I got three for a single confirmed invoice. Will put this into draft for that reason.

Reply:
> After the last listener has been removed from an EventEmitter object, the subscription is released and subsequent attempts to add a listener will result in an error. **Call subscribeTo... again and add new listeners to the new EventEmitter object.**

-- https://www.npmjs.com/package/ln-service#subscriptions

Since I am not sure if we might lose the connection to LND once in a while, I am going to wrap this code with some reconnect logic to be sure (contrary to what I stated in my previous comment).